src/test/transaction_tests.cpp

```diff
@@ -14,143 +15,172 @@
 using namespace std;
 
 // In script_tests.cpp
-extern UniValue read_json(std::string& filename);
+extern UniValue read_json(const std::string& jsondata);
 extern CScript ParseScript(string s);
 
-// Gridcoin, 2017-03-18: Temporarily disable broken tests.
-// Possibly missing Gridcoin fields in JSON data.
-#if 0
+// In init.cpp
+extern void InitLogging();
+
+namespace {
+std::atomic<bool> logging_initialized = false;
+
+// We need to initialize logging with output to console so that error() messages can be seen from the test module. This
+// has an atomic flag so that InitializeLogging() actually only initializes the first time it is called in the tests, and
+// the rest is a no-op, since the logger is a singleton.
+void InitializeLogging()
+{
+    if (logging_initialized) return;
+
+    gArgs.ForceSetArg("-debuglogfile", "none");
+    gArgs.ForceSetArg("-printtoconsole", "1");
+
+    InitLogging();
+
+    logging_initialized = true;
+}
+} // anonymous namespace
```

Comment:
This could be handled in `src/test/test_gridcoin.cpp`.

Reply:
I agree with that. I will move it.